### PR TITLE
updates for scikit-learn 1.2.0

### PR DESCRIPTION
**Pull request by mmccarty**
- [x] Update tests and code to handle `scikit-learn==1.2.0` deprecation warnings
- [x] Update environment files
- [x] Update package requirements

### CODE REVIEW

**Comment by mmccarty** [comment]
Ready for review @jrbourbeau cc @betatim

File: dask_ml/cluster/k_means.py [hunk 1]
```diff
@@ -74,6 +74,17 @@ class KMeans(TransformerMixin, BaseEstimator):
         If None, the random number generator is the RandomState instance used
         by `np.random`.
 
+    n_init : 'auto' or int, default=10
+        Number of time the k-means algorithm will be run with different
```
Comment (betatim) [inline dask_ml/cluster/k_means.py:78]:
```suggestion
        Number of times the k-means algorithm will be run with different
```

File: dask_ml/cluster/k_means.py [hunk 2]
```diff
@@ -315,6 +331,16 @@ def k_init(
     oversampling_factor : int, optional
         Only used for ``init='k-means||`''. Controls the additional number of
         candidate centers in each iteration.
+    n_init : 'auto' or int, default=10
+        Number of time the k-means algorithm will be run with different
```
Comment (betatim) [inline dask_ml/cluster/k_means.py:335]:
```suggestion
        Number of times the k-means algorithm will be run with different
```

**Comment by betatim** [comment]
The changes looks reasonable to me. Only question I have is regarding backwards compatibility and such. Does dask-ml make promises about changes of parameter default values?

**Review by betatim — APPROVED** [review]

File: dask_ml/decomposition/pca.py [hunk 3]
```diff
@@ -327,11 +327,11 @@ def _fit(self, X):
         # Compute noise covariance using Probabilistic PCA model
         # The sigma2 maximum likelihood (cf. eq. 12.46)
         self.noise_variance_ = 0.0
-        if n_components < min(self.n_features_, self.n_samples_):
+        if n_components < min(self.n_features_in_, self.n_samples_):
             if solver == "randomized":
                 self.noise_variance_ = (
                     total_variance - self.explained_variance_.sum()
-                ) / (min(self.n_features_, self.n_samples_) - n_components)
+                ) / (min(self.n_features_in_, self.n_samples_) - n_components)
```
Comment (stsievert) [inline dask_ml/decomposition/pca.py:334]: Should this renaming of `n_features_` to `n_features_in_` trigger a deprecation warning?

File: dask_ml/preprocessing/_encoders.py [hunk 4]
```diff
@@ -116,22 +116,29 @@ def __init__(
         categorical_features: Optional[pd.Categorical] = None,
         categories: Union[str, ArrayLike] = "auto",
         drop: Optional[bool] = None,
-        sparse: bool = True,
+        sparse_output: bool = True,
```
Comment (stsievert) [inline dask_ml/preprocessing/_encoders.py:119]: Should  this renaming of `sparse` to `sparse_output` trigger a deprecation warning?

**Comment by mmccarty** [comment]
> The changes looks reasonable to me. Only question I have is regarding backwards compatibility and such. Does dask-ml make promises about changes of parameter default values?

No promises that I'm aware of. From my experience, the project seems to follow scikit-learn for API changes. I don't intend on adding backwards compatibility with older version of scikit-learn in this PR. Others are welcome to follow up.

File: dask_ml/decomposition/pca.py (hunk 3 above)
Comment (mmccarty) [reply dask_ml/decomposition/pca.py:334]: scikit-learn is handling that. In this case, `sklearn.preprocessing.OneHotEncoder` will raise a warning.

File: dask_ml/preprocessing/_encoders.py (hunk 4 above)
Comment (mmccarty) [reply dask_ml/preprocessing/_encoders.py:119]: `sklearn.preprocessing.OneHotEncoder` will raise a warning.

**Comment by TomAugspurger** [comment]
In general, my policy was to match the lowest-supported version of
scikit-learn, but to pretty aggressively bump that minimum supported
version when it became at all challenging to support both versions. If
pople are able to upgrade their version of scikit-learn to the latest, then
they ought to be able to upgrade dask-ml as well (and vice-versa).

**Comment by mmccarty** [comment]
I'm not opposed to this. #957 requires 1.1.3 and is merged. Added #961 to track.

**Review by betatim — APPROVED** [review]

**Comment by mmccarty** [comment]
@jrbourbeau - Any comments here?

**Review by jrbourbeau — COMMENTED** [review]
Apologies for the delayed response. I've not had a change to look at the changes here yet and probably won't be able to get to it for a bit. If you and @betatim are happy with the changes here, I'd recommend moving forward with them. No need to block on me

**Comment by betatim** [comment]
Let's merge this then. I would have clicked the button, but I can't :D

**Pull request merged**